sierra2mlir/src/types/mod.rs

```diff
@@ -93,6 +93,77 @@ impl<'ctx> Compiler<'ctx> {
                         SierraType::Struct { ty: struct_type, field_types: types },
                     );
                 }
+                "Enum" => {
+                    // https://mapping-high-level-constructs-to-llvm-ir.readthedocs.io/en/latest/basic-constructs/unions.html#tagged-unions
+
+                    // in case the enum has variants with the same type, just store a uniqued one.
+                    let mut unique_enum_types = HashSet::new();
+                    // all the enum variant types, even repeated, for later use when storing the variant value.
+                    let mut enum_variant_types = vec![];
+
+                    let _user_type = match &type_decl.long_id.generic_args[0] {
+                        GenericArg::UserType(x) => x,
+                        _ => {
+                            unreachable!("first arg on struct libfunc should always be a user type")
+                        }
+                    };
+
+                    for gen_arg in type_decl.long_id.generic_args.iter().skip(1) {
+                        let gen_arg_type_id = match gen_arg {
+                            GenericArg::Type(gen_arg_typeid) => gen_arg_typeid.id.to_string(),
+                            GenericArg::UserType(user_type_id) => user_type_id.id.to_string(),
```

Comment:
Since the first argument is skipped, this match should only need an arm for `Type` and one that's marked unreachable